crates/swc_css_modules/src/css_modules.rs

```diff
@@ -0,0 +1,266 @@
+use std::collections::HashSet;
+
+use swc_atoms::JsWord;
+use swc_common::{input::SourceFileInput, sync::Lrc, FileName, SourceMap};
+use swc_css_ast::{
+    ComplexSelector, ComplexSelectorChildren, CompoundSelector, PseudoClassSelectorChildren,
+    SelectorList, SubclassSelector, Token, TokenAndSpan,
+};
+use swc_css_parser::{
+    lexer::Lexer,
+    parser::{Parser, ParserConfig},
+};
+use swc_css_visit::{Fold, FoldWith};
+
+use crate::hash::StyleHash;
+
+#[derive(Default)]
+pub struct CssModuleComponent {
+    pub hash_prefix: String,
+    pub class_collect: HashSet<String>,
+    pub group_pseudo_ident: Vec<Vec<String>>,
+}
+
+#[derive(Default)]
+pub struct ChildCssModuleComponent {
+    pub selector_list: Option<SelectorList>,
+}
+
+impl CssModuleComponent {
+    ///
+    /// init css-module context
+    pub fn new(filepath: &str, content: &str) -> Self {
+        CssModuleComponent {
+            hash_prefix: StyleHash::generate_css_module_hash(filepath, content),
+            class_collect: Default::default(),
+            group_pseudo_ident: vec![],
+        }
+    }
+
+    pub fn convert_token_and_span(t: TokenAndSpan) -> Result<String, String> {
+        let res: String;
+        if let Token::Ident { value, .. } = t.token {
+            res = value.to_string();
+        } else if let Token::Hash { value, .. } = t.token {
+            res = value.to_string();
+        } else if let Token::AtKeyword { value, .. } = t.token {
+            res = value.to_string();
+        } else if let Token::WhiteSpace { value } = t.token {
+            res = value.to_string();
+        } else if let Token::Number { value, .. } = t.token {
+            res = value.to_string();
+        } else if let Token::Percentage { value, .. } = t.token {
+            res = value.to_string();
+        } else if let Token::Url { value, .. } = t.token {
+            res = value.to_string();
+        } else if let Token::String { value, .. } = t.token {
+            res = value.to_string();
+        } else if let Token::BadString { value, .. } = t.token {
+            res = value.to_string();
+        } else if let Token::Delim { value, .. } = t.token {
+            res = value.to_string();
+        } else {
+            return Err("convert token has no matching correct type".to_string());
+        }
+        if !res.is_empty() {
+            Ok(res)
+        } else {
+            Err("convert token has error".to_string())
+        }
+    }
```

Comment:
We should avoid it, better to add the `css_modules` option for CSS parser and parse `:global`/`:local` content as selectors, it will be faster and more easy